http/org.wso2.carbon.transport.http.netty/src/test/java/org/wso2/carbon/transport/http/netty/contentaware/listeners/RequestResponseCreationListener.java

```diff
@@ -0,0 +1,159 @@
+        executor.execute(() -> {
+            try {
+                int length = httpRequest.getFullMessageLength();
+                List<ByteBuffer> byteBufferList = httpRequest.getFullMessageBody();
+                ByteBuffer byteBuffer = ByteBuffer.allocate(length);
+                byteBufferList.forEach(byteBuffer::put);
+                String requestValue = new String(byteBuffer.array());
+                byte[] arry = responseValue.getBytes("UTF-8");
+
+                HTTPCarbonMessage newMsg = httpRequest.cloneCarbonMessageWithOutData();
+                if (newMsg.getHeader(Constants.HTTP_TRANSFER_ENCODING) == null) {
+                    newMsg.setHeader(Constants.HTTP_CONTENT_LENGTH, String.valueOf(arry.length));
+                }
+                ByteBuffer byteBuffer1 = ByteBuffer.allocate(arry.length);
+                byteBuffer1.put(arry);
+                byteBuffer1.flip();
+                newMsg.addHttpContent(new DefaultLastHttpContent(Unpooled.wrappedBuffer(byteBuffer1)));
+                newMsg.setProperty(Constants.HOST, TestUtil.TEST_HOST);
+                newMsg.setProperty(Constants.PORT, TestUtil.TEST_HTTP_SERVER_PORT);
+
+                Map<String, Object> transportProperties = new HashMap<>();
+                Set<TransportProperty> transportPropertiesSet = configuration.getTransportProperties();
+                if (transportPropertiesSet != null && !transportPropertiesSet.isEmpty()) {
+                    transportProperties = transportPropertiesSet.stream().collect(
+                            Collectors.toMap(TransportProperty::getName, TransportProperty::getValue));
+
+                }
+
+                String scheme = (String) httpRequest.getProperty(Constants.PROTOCOL);
+                SenderConfiguration senderConfiguration = HTTPConnectorUtil
+                        .getSenderConfiguration(configuration, scheme);
+
+                HttpWsConnectorFactory httpWsConnectorFactory = new HttpWsConnectorFactoryImpl();
+                HttpClientConnector clientConnector =
+                        httpWsConnectorFactory.createHttpClientConnector(transportProperties, senderConfiguration);
+
+                HttpResponseFuture future = clientConnector.send(newMsg);
+                future.setHttpConnectorListener(new HttpConnectorListener() {
+                    @Override
+                    public void onMessage(HTTPCarbonMessage httpResponse) {
+                        executor.execute(() -> {
+                            int length = httpResponse.getFullMessageLength();
+                            List<ByteBuffer> byteBufferList = httpResponse.getFullMessageBody();
+
+                            ByteBuffer byteBuffer = ByteBuffer.allocate(length);
+                            byteBufferList.forEach(byteBuffer::put);
+                            String responseValue = new String(byteBuffer.array()) + ":" + requestValue;
+                            byte[] array = new byte[0];
```

Comment:
initialize this to null, (if we're not throwing an exception in the try/catch later), or else, do not initialize the value here.